test/prog/dftb+/sockets/H2O_cluster/prerun.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+from __future__ import print_function
+import struct
+import socket
+import os
+import numpy as np
+import numpy.linalg as la
+
+# Expecting two geometry steps of communication with DFTB+
+NR_STEPS = 5
+
+def connect():
+    server_address = '/tmp/ipi_dftb'
+
+    # Make sure the socket does not already exist
+    try:
+        os.unlink(server_address)
+    except OSError:
+        if os.path.exists(server_address):
+            raise
+
+    serversocket = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    serversocket.bind(server_address)
+    serversocket.listen(1) # become a server socket, maximum 1 connections
+    connection, address = serversocket.accept()
+
+    return connection
+
+
+def frac2cart(latvecs,coords):
+    newcoords = np.array(coords)
+    for iAt in range(len(coords)):
+        newcoords[iAt] = np.dot(np.transpose(latvecs),coords[iAt])
+    return (newcoords)
+
+
+def readgen(fname):
```

Comment:
We have now `frac2cart`, `readgen` and `receive_all` and `a0` 3 times identically in various prerun scripts. What about creating a `socketutils` module (maybe in the directory above?) and symlink it from each socket test dir, so that it can be easily imported in the respective python scripts?